src/Microsoft.AspNetCore.Routing.Abstractions/RouteData.cs

```diff
@@ -51,6 +51,20 @@ public RouteData(RouteData other)
             }
         }
 
+        /// <summary>
+        /// Creates a new <see cref="RouteData"/> instance with the specified values.
+        /// </summary>
+        /// <param name="values">The <see cref="RouteValueDictionary"/> copied for values.</param>
+        public RouteData(RouteValueDictionary values)
+        {
+            if (values == null)
+            {
+                throw new ArgumentNullException(nameof(values));
+            }
+
+            _values = new RouteValueDictionary(values);
```

Comment:
This copies like `RouteData` does. The downside is changes in one RVD won't be automatically reflected in the other.

Reply:
This seems kinda broken no? What if someone changes the values during the request?

Reply:
I don't know. Is changing them during a request a thing people do? Do we want changes in the back-compat dictionary effecting endpoints? What do you think should happen here?

Reply:
Yes, people definitely do things to the route value dictionary. I think it will catch some people out if there are two copies.